src/context/WalletProvider/KeepKey/components/Connect.tsx

```diff
@@ -113,7 +113,10 @@ export const KeepKeyConnect = () => {
           connectedType: KeyManager.KeepKey,
         },
       })
-      dispatch({ type: WalletActions.SET_IS_CONNECTED, payload: true })
+      dispatch({
+        type: WalletActions.SET_IS_CONNECTED,
+        payload: { isConnected: true, modalType: state.modalType },
```

Comment:
Allows us to set the connected modalType, vs. the `modalType` (the one at modal time). Now that I'm writing this, there is probably a way to access this already and could potentially revert these?

Reply:
You are using the `modalType` from `state`, then resetting this? It doesn't look like making sense

Reply:
Nice catch! For future reference, `SET_IS_CONNECTED` wasn't doing anything with `modalType`. 


Sanity spot check: https://jam.dev/c/a21a0a93-3378-4ec2-8515-7f965cf3555f

bdba8a5a39e86c08dc18d51135ea62b1cd76825f